### PR TITLE
Map processors

### PR DESCRIPTION
**Pull request by jrief**
This PR replaces PR map-processors which did not merge

### CODE REVIEW

**Comment by jrief** [comment]
Any idea why the unit test fails.
I didn't touch that code, neither the exception emerges from there.

**Comment by ojii** [comment]
https://travis-ci.org/ojii/django-sekizai/jobs/49452709

I recommend running the tests locally (make a venv, install django-classy-tags, django and pep8. then do ``python runtests.py``).

**Comment by jrief** [comment]
I already did that and all unit tests pass locally.

**Comment by ojii** [comment]
Running your branch locally:

```
(env) jonas:~/workspace/django-sekizai(jrief-map-processors2)$ python runtests.py 
Creating test database for alias 'default'...
..............................F........
======================================================================
FAIL: test_pep8 (sekizai.tests.SekizaiTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jonas/workspace/django-sekizai/sekizai/tests.py", line 200, in test_pep8
    "Code not PEP8 compliant:\n{0}".format(errors)
AssertionError: 5 != 0 : Code not PEP8 compliant:
/home/jonas/workspace/django-sekizai/sekizai/tests.py:385:21: E126 continuation line over-indented for hanging indent
/home/jonas/workspace/django-sekizai/sekizai/tests.py:394:21: E126 continuation line over-indented for hanging indent
/home/jonas/workspace/django-sekizai/sekizai/templatetags/sekizai_tags.py:50:80: E501 line too long (96 > 79 characters)
/home/jonas/workspace/django-sekizai/sekizai/templatetags/sekizai_tags.py:51:80: E501 line too long (98 > 79 characters)
/home/jonas/workspace/django-sekizai/sekizai/templatetags/sekizai_tags.py:164:80: E501 line too long (93 > 79 characters)


----------------------------------------------------------------------
Ran 39 tests in 0.140s

FAILED (failures=1)
Destroying test database for alias 'default'...
```

**Comment by jrief** [comment]
```
MacBookPro:django-sekizai jrief$ python runtests.py
Creating test database for alias 'default'...
......................................
----------------------------------------------------------------------
Ran 38 tests in 0.051s

OK
Destroying test database for alias 'default'...
```

;)

**Comment by jrief** [comment]
strange my test runner runs only 38 tests

**Comment by ojii** [comment]
is it up-to-date with master?

**Comment by jrief** [comment]
> is it up-to-date with master?

sure, its build upon commit d5684b81980b1cf7d121845b97c419b562c66140 (Version 0.8.1)

**Comment by ojii** [comment]
latest master is ed986d2eb3963bcfae2f26a4ae38b8391d9bdcab

**Comment by jrief** [comment]
all tests passed except 62.17. This is not by a programming error, but by some stream reset during download. So please consider this commit as passed.

File: sekizai/templatetags/sekizai_tags.py [hunk 1]
```diff
@@ -141,9 +157,14 @@ def render_tag(self, context, name, strip, preprocessor, nodelist):
...
         if strip:
             rendered_contents = rendered_contents.strip()
-        if preprocessor:
+        if preprocessor == 'none':
+            pass
+        elif preprocessor:
             func = import_processor(preprocessor)
             rendered_contents = func(context, rendered_contents, name)
+        elif name in _mapped_preprocessors:
+            rendered_contents = \
```
Comment (ojii) [inline sekizai/templatetags/sekizai_tags.py]: nitpick, but I really dislike using backslash as line-continuation.

Could you change this to:

```python
rendered_contents = _mapped_preprocessors[name](
    context,
    rendered_contents,
    name
)
```

File: sekizai/templatetags/sekizai_tags.py [hunk 2]
```diff
@@ -41,6 +41,18 @@ def import_processor(import_path):
     return getattr(module, object_name)
 
 
+def import_mapped_processors(sekizai_processors):
+    mapped_processors = {}
+    for name, processor in sekizai_processors.items():
+        mapped_processors[name] = import_processor(processor)
+    return mapped_processors
+
+_mapped_preprocessors = import_mapped_processors(
```
Comment (ojii) [inline sekizai/templatetags/sekizai_tags.py]: Is there no way we can get rid of these at-import globals? Could this move inside the template tag classes (``__init__`` or ``render`` or somewhere)?

**Pull request closed without merging**